lib/postgres/postgres-pr/message.rb

```diff
@@ -46,7 +46,14 @@ def self.read(stream, startup=false)
     type = stream.read_exactly_n_bytes(1) unless startup
     length = stream.read_exactly_n_bytes(4).to_s.unpack('N').first  # FIXME: length should be signed, not unsigned
 
-    raise ParseError unless length >= 4
+    raise ParseError if (!length.nil? || length >= 4)
```

Comment:
I don't understand the `!` in there?  I thought we needed to raise a parse error if the value was nil?

Reply:
Also, shouldn't the second conditional change to `length < 3`?

Reply:
Ah sorry I forgot to convert these conditions when changing the statement from `unless` to `if`. Your perfectly right both of these should have changed.

Reply:
Also technically it should be `length < 4`